vhd/packer/configure-windows-vhd.ps1

```diff
@@ -114,8 +114,9 @@ function Get-FilesToCacheOnVHD {
             "https://kubernetesartifacts.azureedge.net/kubernetes/v1.21.0-beta.0/windowszip/v1.21.0-beta.0-1int.zip"
         );
         "c:\akse-cache\win-vnet-cni\" = @(
-            "https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.0_hotfix/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.0_hotfix.zip",
-            "https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.2/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.2.zip"
+            "https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.2/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.2.zip",
+            "https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.2/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.7.zip"

```

Comment:
azure-cni/v1.2.2/binaries
=>
azure-cni/v1.2.7/binaries

Reply:
I'm really surprised the VHD builds for windows didn't fail because of this.

Reply:
do you mean the E2E tests should have failed because of what 1.2.7 fixes?

Reply:
The URL is incorrect 
https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.2/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.7.zip 
should be 
https://kubernetesartifacts.azureedge.net/azure-cni/v1.2**.7**/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.7.zip

Reply:
I would have expected this to fail during the VHD build

Reply:
Hmm, looks like this might be a regression that got introduced when we switched to using curl.exe.

```cmd
curl -L https://kubernetesartifacts.azureedge.net/azure-cni/v1.2.2/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.2.7.zip
﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>BlobNotFound</Code><Message>The specified blob does not exist.
RequestId:786d8a4f-601e-00df-2d04-168166000000
Time:2021-03-10T23:26:09.4129514Z</Message></Error>
C:\Users\marosset>echo %errorlevel%
0
```

curl.exe is returning success here...

Reply:
AgentBaker uses https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/test/windows-vhd-content-test.ps1 to check the downloaded file

Reply:
looks like curl doesn't exit non zero on 404s and others: https://superuser.com/a/657174

Reply:
Yea, it looks like if you pass `-f` it'll exit 22 for 404s.
Let me update this and other usage to check $LASTEXITCODE and error.